packages/vue-output-target/vue-component-lib/utils.ts

```diff
@@ -91,8 +91,17 @@ export const defineContainer = <Props, VModelType = string | number | boolean>(
         const eventsNames = Array.isArray(modelUpdateEvent) ? modelUpdateEvent : [modelUpdateEvent];
         eventsNames.forEach((eventName: string) => {
           el.addEventListener(eventName.toLowerCase(), (e: Event) => {
-            modelPropValue = (e?.target as any)[modelProp];
-            emit(UPDATE_VALUE_EVENT, modelPropValue);
+            /**
+             * Only update the v-model binding if the event's target is the element we are
+             * listening on. For example, Component A could emit ionChange, but it could also
+             * have a descendant Component B that also emits ionChange. We only want to update
+             * the v-model for Component A when ionChange originates from that element and not
+             * when ionChange bubbles up from Component B.
+             */
+            if (e.target.tagName === el.tagName) {
```

Comment:
Is it guaranteed that at least one of these two values can never be `undefined`? My only thought here is running into a case where we end up doing:
```
if (undefined === undefined) {
```

Reply:
Good question. `Element.tagName` is [always defined](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgGxgXhgExMArgWzmKAOmACc4BDKOAUQTjwIAoByNASwDdmBKAbgCh+iQlHIBzAHLk8QA). 

If `e.target` is defined then it should be an `Element` (which means `tagName` is defined). However, `EventTarget` can [potentially be `null`](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgNxgXhmOB3GBhArtEAWwFEE4woAKAcgDMQRqBKAbgCg3EA6KAQwCcA5nChA). My understanding is that `target` can be null when an event is created but not dispatched on an element. You can test this by logging `new CustomEvent('foo')`. In this case, the event is dispatched on an element, so `target` should be non-null.